octoprint_PrintTimeGenius/static/js/PrintTimeGenius.js

```diff
@@ -59,30 +61,43 @@ $(function() {
     });
     self.printerStateViewModel.printTimeLeftOrigin.valueHasMutated();
 
-    self.recurseFiles = function(items) {
-      let results = [];
-      for (let item of items) {
-        if (item["type"] == "machinecode") {
-          results.push(item);
-        }
-        if ("children" in item) {
-          results = results.concat(self.recurseFiles(item.children));
+    self.theFiles = function(items) {
+    	let results = [];
+    	let queue = [{children: items}];
+
+    	while (queue.length > 0) {
+    		item = queue.shift();
+    		results.push(...item.children.filter(item => item["type"] == "machinecode"));
+    		queue.push(...item.children.filter(item => "children" in item));
+    	}
+    	return results;
+    };
+
+    self.update_filelist = function() {
+        let files = self.theFiles(self.filesViewModel.allItems())
+            .sort(function(a,b) {
+                if (_.has(a, "gcodeAnalysis.progress") != _.has(b, "gcodeAnalysis.progress")) {
+                    return (_.has(a, "gcodeAnalysis.progress") - _.has(b, "gcodeAnalysis.progress"));
+                }
+                return a.path.localeCompare(b.path);
+            });
+        self.FileList(files);
+    };
+
+    self.onEventUpdatedFiles = function(payload) {
+        if (self.settings_visible() && payload.type === 'printables') {
+            setTimeout(self.update_filelist, 20000);
         }
-      }
-      return results;
-    }
+    };
 
-    self.FileList = ko.pureComputed(function() {
-      return self.recurseFiles(self.filesViewModel.allItems()).slice()
-          .sort(function(a,b) {
-            if (_.has(a, "gcodeAnalysis.progress") !=
-                _.has(b, "gcodeAnalysis.progress")) {
-              return (_.has(a, "gcodeAnalysis.progress") -
-                      _.has(b, "gcodeAnalysis.progress"));
-            }
-            return a.path.localeCompare(b.path);
-          });
-    });
+    self.onSettingsShown = function() {
+        self.settings_visible(true);
+        self.update_filelist();
+    };
+
+    self.onSettingsHidden = function() {
+        self.settings_visible(false);
```

Comment:
This line should stay the same.  When it run, it, too, will trigger `self.FileList` `ko.pureComputed` to run again.  However, it'll return immediately because the value is false.